src/flask_principal.py

```diff
@@ -11,6 +11,9 @@
 """
 
 from __future__ import with_statement
+from typing import Optional
+from typing import Callable
+import typing_extensions
```

Comment:
All tests are failing because they cannot find the `typing_extensions` library. It does not appear that the library was added to the dependencies. Also, look at what value this additional dependency adds.